src/utils/useScrollDebounce.ts

```diff
@@ -0,0 +1,51 @@
+import { useEffect, useRef } from 'react';
+
+interface Position {
+  x: number;
+  y: number;
+}
+
+const getScrollPosition = (): Position => {
+  const scrollPosition = document.body.getBoundingClientRect();
+
+  return {
+    x: scrollPosition.x || 0,
+    y: scrollPosition.y || 0,
+  };
+};
+
+/**
+ * Detects scroll events in the window and calls the effect with the scroll
+ * position when it has settled for longer than the debounceTime (ms).
+ */
+const useScrollDebounce = (
```

Comment:
This hook was adapted from https://github.com/n8tb1t/use-scroll-position, which allows for a time increment on checking scroll position, but not debouncing.